### PR TITLE
Added PrepSRL

### PR DESCRIPTION
**Pull request by christos-c**
- Created depparse-based Edison transformers (used for PrepSRL features)
- Small fix in depparse

The model file is here: https://s3.us-east-2.amazonaws.com/com.christos-c.share/prepsrl-models.jar
Snippet pasted in for convenience:
```
deploy:deploy-file -Dfile=prepsrl-models.jar -DgroupId=edu.illinois.cs.cogcomp -DartifactId=illinois-prepsrl-model -Dversion=3.0 -Dpackaging=jar  -Durl=scp://bilbo.cs.illinois.edu:/mounts/bilbo/disks/0/www/cogcomp/html/m2repo -DrepositoryId=CogcompSoftware
```

### CODE REVIEW

**Comment by danyaljj** [comment]
Uploaded: http://cogcomp.cs.illinois.edu/m2repo/edu/illinois/cs/cogcomp/illinois-prepsrl-model/

**Comment by mssammon** [comment]
looks like you need to add license headers...

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotator.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import edu.illinois.cs.cogcomp.core.utilities.configuration.ResourceManager;
+import edu.illinois.cs.cogcomp.depparse.DepConfigurator;
+import edu.illinois.cs.cogcomp.lbjava.classify.Classifier;
+import edu.illinois.cs.cogcomp.prepsrl.data.DataReader;
+import edu.illinois.cs.cogcomp.prepsrl.data.PrepSRLDataReader;
+import edu.illinois.cs.cogcomp.prepsrl.inference.ConstrainedPrepSRLClassifier;
+
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/PrepSRLAnnotator.java]: Please add a javadoc describing annotation at a suitably high level

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/DataReader.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import edu.illinois.cs.cogcomp.prepsrl.PrepSRLConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The base data reader class containing the `LBJava` {@link Parser} code used by all applications.
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/DataReader.java]: Please indicate whether this is a completely generic base reader or if it imposes some constraints on input (is it just for SRL data?)

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/PrepSRLDataReader.java [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FilenameFilter;
+import java.util.*;
+
+/**
+ * Data reader for the Semeval dataset. Creates the {@link ViewNames#SRL_PREP} view using the
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/PrepSRLDataReader.java]: Please indicate a specific semeval task, OR describe the format/provide a URL to the appropriate documentation.

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/Preprocessor.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import edu.stanford.nlp.pipeline.ParserAnnotator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * An annotation preprocessor used by all the modules, an instance of. The configurations parameters
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/data/Preprocessor.java]: all which modules? an instance of what?

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/ConstrainedPrepSRLClassifier.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import edu.illinois.cs.cogcomp.lbjava.classify.DiscretePrimitiveStringFeature;
+import edu.illinois.cs.cogcomp.lbjava.classify.Feature;
+import edu.illinois.cs.cogcomp.lbjava.classify.FeatureVector;
+import edu.illinois.cs.cogcomp.lbjava.infer.InferenceManager;
+import edu.illinois.cs.cogcomp.prepsrl.PrepSRLConfigurator;
+import edu.illinois.cs.cogcomp.prepsrl.PrepSRLClassifier;
+
+import java.io.File;
+
+public class ConstrainedPrepSRLClassifier extends Classifier {
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/ConstrainedPrepSRLClassifier.java]: Javadoc: description

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/PrepSRLInference.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.infer.ilp.OJalgoHook;
+import edu.illinois.cs.cogcomp.lbjava.infer.ILPInference;
+import edu.illinois.cs.cogcomp.lbjava.learn.Learner;
+import edu.illinois.cs.cogcomp.lbjava.learn.Normalizer;
+import edu.illinois.cs.cogcomp.lbjava.learn.Softmax;
+import edu.illinois.cs.cogcomp.prepsrl.inference.constraints.PrepSRLInferenceConstraints;
+
+public class PrepSRLInference extends ILPInference {
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/PrepSRLInference.java]: javadoc: description

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/constraints/LegalRoles.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.core.utilities.configuration.ResourceManager;
+import edu.illinois.cs.cogcomp.lbjava.infer.*;
+import edu.illinois.cs.cogcomp.prepsrl.PrepSRLConfigurator;
+import edu.illinois.cs.cogcomp.prepsrl.PrepSRLClassifier;
+import edu.illinois.cs.cogcomp.prepsrl.data.PrepSRLDataReader;
+
+import java.io.File;
+
+class LegalRoles extends ParameterizedConstraint {
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/constraints/LegalRoles.java]: javadoc: description

File: prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/constraints/PrepSRLInferenceConstraints.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+ * Developed by: The Cognitive Computation Group University of Illinois at Urbana-Champaign
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.prepsrl.inference.constraints;
+
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.lbjava.infer.FirstOrderConstraint;
+import edu.illinois.cs.cogcomp.lbjava.infer.ParameterizedConstraint;
+
+public class PrepSRLInferenceConstraints extends ParameterizedConstraint {
```
Comment (mssammon) [inline prepsrl/src/main/java/edu/illinois/cs/cogcomp/prepsrl/inference/constraints/PrepSRLInferenceConstraints.java]: Javadoc: description

**Comment by danyaljj** [comment]
for license header: `mvn license:format`

**Pull request merged**